src/app/home/pages/PlansPage/components/AnalyticsTable.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { useEffect, useState } from 'react';
+import { Table, TableBody, TableHeader } from '@patternfly/react-table';
+import {
+  Button,
+  Bullseye,
+  EmptyState,
+  Title,
+  Text,
+  TextContent,
+  TextVariants,
+  Spinner,
+  Popover,
+  EmptyStateIcon,
+  EmptyStateBody,
+  EmptyStateVariant,
+  PopoverPosition,
+  Progress,
+  ProgressSize,
+  ProgressVariant,
+  Tooltip,
+  TooltipPosition,
+} from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { IAnalytic } from '../../../../plan/duck/types';
+import ReactJson from 'react-json-view';
+const styles = require('./AnalyticsTable.module');
+
+interface IProps {
+  migAnalytics?: IAnalytic[];
+  id?: string;
+  type?: string;
+  isPlanLocked: boolean;
+  isRefreshingAnalytic: boolean;
+  latestAnalytic: IAnalytic;
+  analyticPercentComplete: number;
+}
+
+const AnalyticsTable: React.FunctionComponent<IProps> = ({
+  migAnalytics,
+  isPlanLocked,
+  latestAnalytic,
+  analyticPercentComplete,
+  isRefreshingAnalytic,
+}) => {
+  const [currentRows, setCurrentRows] = useState([]);
+  const [isLoadingAnalytics, setIsLoadingAnalytics] = useState<boolean>(false);
+
+  const columns = [
+    { title: 'Namespace' },
+    { title: 'Kubernetes resources' },
+    { title: 'PVCs' },
+    { title: 'PVC capacity' },
+    { title: 'Images' },
+    { title: 'Image size' },
+    { title: '' },
+  ];
+
+  function ProgressWrapper() {
+    return (
+      <Progress
+        value={analyticPercentComplete ? analyticPercentComplete : 0}
+        title={'Retrieving namespace details'}
+        size="sm"
+      />
+    );
+  }
+
+  useEffect(() => {
+    const namespaces = latestAnalytic?.status?.analytics?.namespaces
+      ? latestAnalytic?.status?.analytics?.namespaces
+      : [];
+
+    setIsLoadingAnalytics(
+      (analyticPercentComplete !== 100 && latestAnalytic) || isRefreshingAnalytic ? true : false
+    );
+
+    const mappedRows = namespaces.map((namespace) => {
```

Comment:
Looks like this could blow up on a map call because `namespaces` could be null from a short-circuit return on 70. Do we need to default mapped rows to an empty state if namespaces is null?

It actually feels like with the introduction of this syntax it's really easy to ignore what happens if that returns a null and creates a lot of opportunity for calls on a null value, if I'm understanding that correct?

Reply:
@eriknelson If you look above on line 72, 


```
 const namespaces = latestAnalytic?.status?.analytics?.namespaces
      ? latestAnalytic?.status?.analytics?.namespaces
      : [];
```

I think that should take care of any null values.

Reply:
Yep, you're right. Mssed that.